packages/conf/__typetests__/index.test-d.tsx

```diff
@@ -33,13 +33,13 @@ expectAssignable<LinguiConfig>({
     tsExperimentalDecorators: false,
   },
   fallbackLocales: {} as FallbackLocales,
-  format: "po",
+  format: "po" as const,
```

Comment:
The inferred type of `"po"` is `string` and the inferred type of `"po" as const` is `"po"`. APIs of `tsd-lite` distinguishes between those two. For my understanding, this is more precise approach.

Reply:
Hi, thanks for help. In this case what we have in test is different from what users will write in theirs code. Isn't the reason of these tests is to check that all user-scenarios would work as much as closer to real life? 

I mean, what if it will pass type tests only because of explicit casting to `const` but will not work in the user's codebase without it?

Reply:
Good point. I see at least two scenarios on user side:

```ts
export const config: LinguiConfig = {
  locales: ["en", "pl"],
  orderBy: "messageId", // no need for `as const`, that’s how `tsd` interprets it
}
```

```ts
let config: LinguiConfig = {
  locales: ["en", "pl"],
}

// simplified for demonstration, this could be returned by some function or so
const extendedConfig = {
  locales: ["en", "pl"],
  orderBy: "origin" as const, // requires `as const` to pass assignability check later
}

config = extendedConfig
```

Through my eyes, first case is declaration of a variable. Assignability happens in the second example. That’s why requiring `as const` makes more sense for me.

Would be good to cover both, or? I can add the declaration case to the type test file. If that is acceptable.

Reply:
Also try this with `tsd`:

```ts
import { LinguiConfig } from "@lingui/conf"
import { expectAssignable, expectNotAssignable } from "tsd"

// pass
expectAssignable<LinguiConfig>({
  locales: ["en", "pl"],
  format: "po",
})

// also pass! wait? how?!
expectNotAssignable<LinguiConfig>({
  locales: ["en", "pl"],
  format: "po",
})
```

Reply:
Ah.. Actually there is no need for extra test case, because the two below are equivalent:

```ts
const format: "po" = "po"

const format = "po" as const
```

Having `"po" as const` in an assignability test means that user will be allowed passing bare `"po"` while declaring a variable, but `"po" as const` will be required in case of assignment.

---

Sorry about this stream of thoughts. Type testing turns me on ;D